.github/workflows/main.yml

```diff
@@ -37,6 +37,28 @@ jobs:
       - uses: actions/checkout@v3
       - run: make podman-build
 
+  test-unit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '1.19'
+      - run: make test
+
+  test-api:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '1.19'
+      - run: |
+          make run &
```

Comment:
@djzager I'm not sure about this part, we might discuss it or add in follow-up PR.